pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go

```diff
@@ -338,15 +340,110 @@ func (pl *VolumeRestrictions) EventsToRegister() []framework.ClusterEventWithHin
+		{Event: framework.ClusterEvent{Resource: framework.PersistentVolumeClaim, ActionType: framework.Add | framework.Update}, QueueingHintFn: pl.isSchedulableAfterPersistentVolumeClaimChange},
 	}
 }
 
+// isSchedulableAfterPodDeleted is invoked whenever a pod deleted,
+// It checks whether the deleted pod will conflict with volumes of other pods on the same node
+func (pl *VolumeRestrictions) isSchedulableAfterPodDeleted(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) framework.QueueingHint {
+	oldPod, _, err := util.As[*v1.Pod](oldObj, newObj)
+	if err != nil {
+		logger.Error(err, "unexpected objects in isSchedulableAfterPodDeleted", "oldObj", oldObj, "newObj", newObj)
+		return framework.QueueAfterBackoff
+	}
+
+	readWriteOncePVCUseByAnotherPod, err := pl.readWriteOncePodPVCUseByAnotherPod(context.TODO(), pod)
+	if err != nil {
+		logger.Error(err, "encountered an error when obtaining PVC according to the pod", "pod", klog.KObj(pod))
+	}
+
+	if pl.enableReadWriteOncePod && readWriteOncePVCUseByAnotherPod {
+		logger.V(4).Info("The Pod uses a PVC with ReadWriteOncePod access mode that is occupied by another Pod.", "pod", klog.KObj(pod))
+		return framework.QueueAfterBackoff
+	}
+
+	nodeInfo := framework.NewNodeInfo(oldPod)
+	if !satisfyVolumeConflicts(pod, nodeInfo) {
+		return framework.QueueAfterBackoff
+	}
+
+	return framework.QueueSkip
+}
+
+// isSchedulableAfterNodeAdded is invoked whenever a node added, It checks whether
+// that change made a previously unschedulable pod schedulable.
+func (pl *VolumeRestrictions) isSchedulableAfterNodeAdded(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) framework.QueueingHint {
+	_, newNode, err := util.As[*v1.Node](oldObj, newObj)
+	if err != nil {
+		logger.Error(err, "unexpected objects in isSchedulableAfterNodeAdded", "oldObj", oldObj, "newObj", newObj)
+		return framework.QueueAfterBackoff
+	}
+
+	readWriteOncePVCUseByAnotherPod, err := pl.readWriteOncePodPVCUseByAnotherPod(context.TODO(), pod)
+	if err != nil {
+		logger.Error(err, "encountered an error when obtaining PVC according to the pod", "pod", klog.KObj(pod))
+	}
+
+	if pl.enableReadWriteOncePod && readWriteOncePVCUseByAnotherPod {
+		logger.V(4).Info("The Pod uses a PVC with ReadWriteOncePod access mode that is occupied by another Pod.", "pod", klog.KObj(pod))
+		return framework.QueueAfterBackoff
+	}
+
+	logger.V(4).Info("node was added", "pod", klog.KObj(pod), "node", klog.KObj(newNode))
+	return framework.QueueSkip
+}
+
+// isSchedulableAfterPersistentVolumeClaimChange is invoked whenever a PersistentVolumeClaim changed, It checks whether
+// that change made a previously unschedulable pod schedulable.
+func (pl *VolumeRestrictions) isSchedulableAfterPersistentVolumeClaimChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) framework.QueueingHint {
+	_, _, err := util.As[*v1.PersistentVolumeClaim](oldObj, newObj)
```

Comment:
An explanation for why I think we don't need the information of the pvc here for the later reviewers:
We need to check the pod's RWOP pvcs here anyway(`readWriteOncePodPVCUseByAnotherPod`). And whether this is a good change or bad change, all this change affects is the result of `readWriteOncePodPVCUseByAnotherPod`. So I think `readWriteOncePodPVCUseByAnotherPod` will contain the check for this change itself.

Reply:
I understand what you're thinking, but here it's just a way to avoid unanticipated events, and when it's not as expected we can output logs to avoid masking the errors

Reply:
Yes, let's leave it for a discussion, for whether this is worth it :)